internal/reminder/publisher.go

```diff
@@ -13,12 +13,33 @@ import (
 	"github.com/rs/zerolog"
 
 	"github.com/mindersec/minder/internal/events/common"
+	natsinternal "github.com/mindersec/minder/internal/events/nats"
+	"github.com/mindersec/minder/pkg/eventer/constants"
 )
 
+func (r *reminder) getMessagePublisher(ctx context.Context) (message.Publisher, common.DriverCloser, error) {
+	switch r.cfg.EventConfig.Driver {
+	case constants.NATSDriver:
+		return r.setupNATSPublisher(ctx)
+	case constants.SQLDriver:
+		return r.setupSQLPublisher(ctx)
+	default:
+		return nil, nil, fmt.Errorf("unknown publisher type: %s", r.cfg.EventConfig.Driver)
+	}
+}
```

Comment:
I'm nervous about having this diverge from `internal/events/events.go`.  Can we use an [`eventer.Interface` from `eventer.New()`](https://pkg.go.dev/github.com/mindersec/minder@v0.0.71/pkg/eventer) here?

(I need to do some refactoring in that package to move `eventer/interface` into `eventer`, but I'll wait until this code is landed to reduce the amount of conflicts.)

Reply:
> I'm nervous about having this diverge from internal/events/events.go

`internal/events/events.go` is all about `serverconfig`, it would diverge from it as this is `reminderconfig`. I don't think we should use the [interfaces.Interface](https://pkg.go.dev/github.com/mindersec/minder@v0.0.71/pkg/eventer/interfaces#Interface):

```go
// Interface is a combination of the Publisher, Registrar, and Service interfaces.
// This is handy when spawning the eventer in a single function for easy setup.
type Interface interface {
	Publisher
	Registrar
	Service
}
```

as reminder only needs publishing capability, I'd be happy to implement [interfaces.Publisher](https://pkg.go.dev/github.com/mindersec/minder@v0.0.71/pkg/eventer/interfaces#Publisher) for `reminder`.

Reply:
I think it's fine to use the `interfaces.Publisher` in the reminder code, but it feels like it might be easiest if the reminder code expected an `interfaces.Publisher` (which is satisfied by the current eventing implementations), but the construction code used `NewEventer` to construct the `interfaces.Publisher`.